Lib/typing.py

```diff
@@ -1429,11 +1430,16 @@ def name_by_id(user_id: UserId) -> str:
         num = UserId(5) + 1     # type: int
     """
 
-    def new_type(x):
-        return x
+    NewTypeClass = type(name, (object,), {
```

Comment:
TBH, I don't really like this. One of the main motivation for `NewType` was to avoid any overhead of creating a new class objects. So this essentially defeats the purpose of `NewType`. If you really think nice `repr` is so important, then you can create a special class _once_ in the `typing` module, and then just instantiate it. For example:
```python
>>> class NewType:
...     def __init__(self, name, tp):
...         self.__name__ = name
...         self.__supertype__ = tp
...     def __call__(self, arg):
...         return arg
...     def __repr__(self):
...         return f"NewType <{self.__name__}:{self.__supertype__.__name__}>"
... 
>>> 
>>> CustomerId = NewType("CustomerId", int)
>>> 
>>> CustomerId
NewType <CustomerId:int>
>>> CustomerId(12)
12
```

Reply:
I’d definitely rather do something like this – presumably with `__slots__` defined for the `__name__` and `__supertype__` values, _n’est ce pas?_ I’ll update the PR accordingly, yes!